Sources/Swift/Tools/LoadValidator.swift

```diff
@@ -0,0 +1,81 @@
+@_implementationOnly import _SentryPrivate
+import Darwin
+import Foundation
+import MachO
+
+@objc
+@_spi(Private) public final class LoadValidator: NSObject {
+    // Any class should be fine, ObjC classes are better
+    static let targetClassName = NSStringFromClass(SentryDependencyContainerSwiftHelper.self)
+    
+    // This function is used to check for duplicated SDKs in the binary.
+    // Since `SentryBinaryImageInfo` is not public and only available through the Hybrid SDK, we use the expanded parameters.
+    @objc
+    @_spi(Private) public class func checkForDuplicatedSDK(imageName: String, imageAddress: NSNumber, imageSize: NSNumber) {
+        internalCheckForDuplicatedSDK(imageName, imageAddress.uint64Value, imageSize.uint64Value,
+                                      objcRuntimeWrapper: SentryDefaultObjCRuntimeWrapper.sharedInstance(),
+                                      dispatchQueueWrapper: SentryDispatchQueueWrapper())
+    }
+    
+    class func internalCheckForDuplicatedSDK(_ imageName: String, _ imageAddress: UInt64, _ imageSize: UInt64, objcRuntimeWrapper: SentryObjCRuntimeWrapper, dispatchQueueWrapper: SentryDispatchQueueWrapper, resultHandler: ((Bool) -> Void)? = nil) {
+        let systemLibraryPath = "/usr/lib/"
+#if targetEnvironment(simulator)
+        let ignoredPath = "/Library/Developer/CoreSimulator/Profiles/Runtimes/"
+#else
+        let ignoredPath = "/System/Library/"
+#endif
+        guard !imageName.contains(ignoredPath) && !imageName.hasPrefix(systemLibraryPath) else {
```

Comment:
`m`: `hasPrefix` should be close to O(1) in most cases while `contains` can be O(n). I think we should be able to use `hasPrefix` here.
```suggestion
        guard !imageName.hasPrefix(ignoredPath) && !imageName.hasPrefix(systemLibraryPath) else {
```